hr_holidays_public/models/hr_leave.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2017-2018 Tecnativa - Pedro M. Baeza
+# Copyright 2018 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class HrLeave(models.Model):
+    _inherit = 'hr.leave'
+
+    def _get_number_of_days(self, date_from, date_to, employee_id):
+        if (self.holiday_status_id.exclude_public_holidays or
+                not self.holiday_status_id):
+            instance = self.with_context(
+                employee_id=employee_id,
+                exclude_public_holidays=True,
+            )
+        else:
+            instance = self
+        return super(HrLeave, instance)._get_number_of_days(
+            date_from,
+            date_to,
+            employee_id,
+        )
+
+    def _onchange_employee_id(self):
+        super()._onchange_employee_id()
+        self._onchange_leave_dates()
```

Comment:
Why is this needed?

Is this a workaround for a known problem in Odoo? If so, put the link to the Odoo issue/PR for reference, but I think this has been solved meanwhile. Any way, very ugly to have method overridden without docstring about what is the reason for overriding it...

Reply:
But is this still needed in v12? You have forgotten to add the decorator `@api.onchange` in this one.

Reply:
super `_onchange_employee_id` already has the decorator, as far as I remember, I don't need to copy it
Regarding if it's needed - looks like no, I'm not 100% sure, yet I think you're correct